runtime/src/accounts_index_storage.rs

```diff
@@ -58,7 +58,7 @@ impl BgThreads {
         in_mem: &[Arc<InMemAccountsIndex<T, U>>],
         threads: usize,
         can_advance_age: bool,
-        exit: &Arc<AtomicBool>,
+        exit: Arc<AtomicBool>,
```

Comment:
While you are right, I do not think these optimizations are worth the extra complexity.
It is easier to just say, "`Arc`s should use by-value semantics".
Using references sometimes, and by-value semantics in other cases, is adding complexity.
It saves two atomic operations, one increment and one decrement.
I think, it might be even impossible to measure the impact, considering that the loop will be doing increments and decrements on the same value.

Reply:
Sold

Reply:
There are some cases when functions take `Arc<AtomicBool>` when they actually can be satisfied with `&AtomicBool`.
Previously, they would take `&Arc<AtomicBool>`.

Dereferencing an `&Arc<AtomicBool>` does not increment the counters.  But passing in an `Arc<AtomicBool>` does.
In this PR the number of changes that are like this is low, I believe.
Though it is a bit tricky to count them all.

I am trying to think of at least partially automated way to find all cases where `Arc<AtomicBool>` can be replaced with just `&AtomicBool`.
Will create a separate PR if I will come up with something.